data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaCustomConsumer.java

```diff
@@ -142,6 +144,23 @@ KafkaTopicConsumerMetrics getTopicMetrics() {
         return topicMetrics;
     }
 
+    private <T> long getReceivedTime(final ConsumerRecord<String, T> consumerRecord) {
```

Comment:
We need some unit testing on this logic. 

Also, you could make a class that handles this and it would be easier to verify. But, I'll leave that to you.

```
class PartitionTracker {
  <T> Instant getReceivedTime(final ConsumerRecord<String, T> consumerRecord) {
    // same logic
  }
}
```

Reply:
I will add unit tests but adding a class seems to be little bit too much. Also, we use "PartitionTracker" for tracking the acks at partition level.